tt_metal/hw/inc/blackhole/noc_nonblocking_api.h

```diff
@@ -347,6 +389,14 @@ inline __attribute__((always_inline)) void noc_fast_atomic_increment(
     bool linked,
     bool posted = false) {
     while (!noc_cmd_buf_ready(noc, cmd_buf));
+    if constexpr (NOC_MODE == DM_DYNAMIC_NOC) {
```

Comment:
this seems odd to me.  I think the noc_non_blocking_api shouldn't have visibility into the global state, that is a higher level concept.  perhaps make this a template arg and have the higher level pass it in?

Reply:
added the template arg, propogated it to dataflow_api.h, where noc_mode will be set to DM_DEDICATED for FW, and set to user_specified for Kernel